actions/publish_pypi/action.yml

```diff
@@ -0,0 +1,118 @@
+name: "Build and publish the distribution to the PyPI server"
+description: "Build and publish PyPi wheel using tox for ML repos e.g. sparseml, compressed-tensors"
+
+inputs:
+  publish_pypi_internal:
+    description: "Publish the distribution to the internal PyPI server"
+    required: true
+
+  publish_pypi:
+    description: "Publish the distribution to the pypi.org"
+    required: true
+
+  built_type:
+    description: "Applies a set of rules depending on the environment. Available values: (dev|release|nightly|custom)"
+    required: true
+
+  custom_package_name:
+    description: "Custom package name could be used along wtih the 'custom' build_type"
+    required: false
+
+outputs:
+  whlname:
+    description: "wheel filename"
+    value: ${{ steps.build.outputs.whlname }}
+  tarname:
+    description: "tar.gz filename"
+    value: ${{ steps.build.outputs.tarname }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Validate input parameters
+      run: |
+        if [ "${{ inputs.publish_pypi }}" == "false" ] && [ "${{ inputs.publish_pypi_internal }}" == "false" ]; then
+          echo "Error: At least one of 'publish_pypi' or 'publish_pypi_internal' must be set to 'true'"
+          exit 1
+        fi
+        if [ "${{ inputs.built_type }}" == "custom" ] && [ "${{ inputs.custom_package_name }}" == "" ]; then
+          echo "Error: If 'built_type' is set to 'custom' the 'custom_package_name' must be specified"
+          exit 1
+        fi
+
+    - name: Install tox
+      run: python3 -m pip install --user tox build
+
+    - name: Build the distribution with tox
+      id: build
+      run: |
+        python3 -m tox -e build
```

Comment:
Before this line, run toml edits for name and version: https://github.com/neuralmagic/guidellm/blob/main/utils/inject_build_props.py#L46C9-L47C45

Reply:
@markurtz I just realized that the tox `build` command already runs `python utils/inject_build_props.py` which is a part of a concrete project. Like, do we still need it right here? In that case we limit all the applications to have this script in that exact folder, isn't it?